src/components/Counter/Counter.js

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { getParentForm } from '../../util';
+
+class Counter extends React.Component {
+  constructor() {
+    super();
+
+    this.handleIncrement = this.handleIncrement.bind(this);
+    this.handleAddItem = this.handleAddItem.bind(this);
+    this.handleToggleAddForm = this.handleToggleAddForm.bind(this);
+
+    this.state = {
+      addFormShown: false,
+    };
+  }
+
+  handleIncrement(itemId) {
+    return () => {
+      this.props.incrementCounterItem(this.props.id, itemId);
+    }
+  }
+
+  handleAddItem(e) {
+    const form = getParentForm(e.currentTarget);
+    if(!form) return;
+
+    const title = form.querySelector('[name=title]').value.trim();
+    const unsanitizedLimit = form.querySelector('[name=limit]').value.trim();
+
+    const limit = Math.abs(+(unsanitizedLimit));
+
+    form.reset()
+
+    this.handleToggleAddForm(false)();
+    this.props.addCounterItem(this.props.id, {title, limit});
+  }
+
+  handleRemoveItem(itemId) {
+    return () => {
+      this.props.removeCounterItem(this.props.id, itemId);
+    }
+  }
+
+  handleToggleAddForm(shown) {
+    return () => {
+      this.setState({addFormShown: shown});
+    }
+  }
+
+  renderAddForm() {
+    return (
+      <div className="App--form">
+        <div className="App--form--content">
+          <form action="#">
+            <input
+              type="text"
+              name="title"
+              placeholder="Add title..."
+            />
+            <input
+              type="number"
+              name="limit"
+              placeholder="Add limit..."
+              min="0"
+            />
+            <button className="App--form--complete" onClick={this.handleAddItem} >
```

Comment:
Do I need the class?

Reply:
Not sure if there is currently css for this class.